test/integration/controllers/shop.test.js

```diff
@@ -2,11 +2,34 @@ const supertest = require('supertest');
 
 describe('shop', () => {
 
+  describe('post /api/shop/listing', () => {
+    it('returns 200 if correct data', async () => {
+
+      sandbox.stub(sails.helpers.sdtdApi, 'executeConsoleCommand').resolves({
+        command: 'listitems',
+        parameters: 'terrDestroyedWoodDebris',
+        result: '    some test item\nListed 1 matching items.\n'
+      });
+
+      const response = await supertest(sails.hooks.http.app)
+        .post('/api/shop/listing')
+        .send({
+          serverId: sails.testServer.id,
+          name: 'some test item',
+          amount: '1',
+          price: 50
+        });
+
+      expect(response.body.name).to.eq('some test item');
+      expect(response.statusCode).to.equal(200);
```

Comment:
Is there any chance it'll return 200 but not find the right item? Do you want to check the db too?

Maybe do a test with multiple returns?

Reply:
Added some more tests :smiley: